### PR TITLE
fix mispellings and add minor corrections

### PR DESCRIPTION
**Pull request by Mistobaan**
The main change of the PR is the change of interface from a single Reader(, isText) to a more (IMHO) go idiomatic `ReadFromBuffer` and `ReadFromString`

### CODE REVIEW

**Comment by alonsovidales** [comment]
Awesome 👍

File: tensorflow/contrib/go/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ also to build, load and run Graphs.
 ## Installation
 
 This package depends on the TensorFlow shared libraries, in order to compile
-this libraries follow the [Installing fromsources](https://www.tensorflow.org/versions/r0.8/get_started/os_setup.html#installing-from-sources)
+this libraries follow the [Installing fromsources](https://www.tensorflow.org/versions/r0.9/get_started/os_setup.html#installing-from-sources)
```
Comment (vrv) [inline tensorflow/contrib/go/README.md]: https://www.tensorflow.org/get_started/os_setup.html#installing-from-sources

use that link and it'll redirect to the latest stable version

File: tensorflow/contrib/go/graph.go [hunk 2]
```diff
@@ -118,16 +118,16 @@ func NewGraph() *Graph {
 // content into a new graph. Use the asText parameter to specify if the graph
 // from the reader is provided in Text format.
 func NewGraphFromReader(reader io.Reader, asText bool) (*Graph, error) {
-	graphStr, err := ioutil.ReadAll(reader)
+	b, err := ioutil.ReadAll(reader)
```
Comment (vrv) [inline tensorflow/contrib/go/graph.go]: 'b' seems like a less useful variable name than graphStr was

**Comment by vrv** [comment]
(ping when ready to review)

**Comment by Mistobaan** [comment]
@vrv I think this PR is good. I tried to add some other things like the go bazel rules to build the package uzing bazel but I will wait for this PR to be merged and the branch to be rebased.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

I have no idea if we even have tests for this or what happens, but trying it anyway.

**Comment by Mistobaan** [comment]
@vrv there are few tests, but I don't think they are linked to the CI system.

**Comment by martinwicke** [comment]
This didn't work, I guess.

**Comment by Mistobaan** [comment]
@martinwicke can you guys rebase origin/go on top of origin/master ? a bunch of PR are trying to catch up the go branch to the master one.

**Comment by martinwicke** [comment]
I rebased the go branch on top of current master.

**Comment by Mistobaan** [comment]
🎉  Thanks ! 🎉

**Comment by Mistobaan** [comment]
PR rebased too.

**Comment by danmane** [comment]
@vrv you've looked at this in the past. should we merge it?

**Comment by vrv** [comment]
I dunno, there are no tests triggered for this branch, so merging might break the branch.  @Mistobaan can fix it I guess if there are problems :)

**Comment by Mistobaan** [comment]
👍

**Pull request merged**